src/emulator/renderer/include/renderer/functions.h

```diff
@@ -23,7 +23,7 @@ bool create(FragmentProgram &fp, State &state, const SceGxmProgram &program, con
 bool create(VertexProgram &vp, State &state, const SceGxmProgram &program, const char *base_path);
 void begin_scene(const RenderTarget &rt);
 void end_scene(Context &context, SceGxmSyncObject *sync_object, size_t width, size_t height, size_t stride_in_pixels, uint32_t *pixels);
-bool sync_state(Context &context, const GxmContextState &state, const MemState &mem, bool enable_texture_cache);
+bool sync_state(Context &context, const GxmContextState &state, const MemState &mem, bool enable_texture_cache, bool log_active_shaders, bool log_uniforms);
```

Comment:
Can't pass the `Config` here? I'd rather have this passed everywhere instead of `bool`s or whatever That way we don't have to keep adding args as config options increase.

Reply:
I think I'll have to move Config out of host then?

Reply:
Yeah, it'd have to be made into a new project, but I'm not sure it's the best approach.

Whatever, keep it this way with the arguments for each option, if it becomes too much of a hassle we'll change it in the future.
In some ways it's better anyway, since it's more explicit and you know which part of the config it uses.